### PR TITLE
MAINT: Fix typos found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Also fix this sentence in the documentation, I guess the word `cast` was missing:
> May give a result equivalent to (the intermediat means no warning is given):
> 
> ```
> arr.astype(np.int64).astype(np.int8)
> ```

### CODE REVIEW

File: numpy/core/src/multiarray/textreading/field_types.h [hunk 1]
```diff
@@ -31,7 +31,7 @@
  *       considered boolean True or False).
  *
  * TODO: Aside from nailing down the above notes, it may be nice to expose
- *       these function publically.  This could allow user DTypes to provide
+ *       these function publicly.  This could allow user DTypes to provide
```
Comment (mattip) [inline numpy/core/src/multiarray/textreading/field_types.h:34]: Apparently both are correct but publicly is more commonly used (by 100 to 1 or so), so let's go with that.

**Review by mattip — APPROVED** [review]
LGTM, anyone else want to look?

**Comment by seberg** [comment]
Thanks @DimitriPapadopoulos and Matti.

**Review by kernel-loophole — COMMENTED** [review]
update

**Pull request merged**